src/main/java/baritone/process/ElytraProcess.java

```diff
@@ -329,4 +349,58 @@ public double placeBucketCost() {
             return COST_INF;
         }
     }
+
+    private static boolean isInBounds(BlockPos pos) {
+        return pos.getY() >= 0 && pos.getY() < 128;
+    }
+
+    private boolean isAtEdge(BlockPos pos) {
+        return ctx.world().isAirBlock(pos.north())
+                || ctx.world().isAirBlock(pos.south())
+                || ctx.world().isAirBlock(pos.east())
+                || ctx.world().isAirBlock(pos.west())
+                // corners
+                || ctx.world().isAirBlock(pos.north().west())
+                || ctx.world().isAirBlock(pos.north().east())
+                || ctx.world().isAirBlock(pos.south().west())
+                || ctx.world().isAirBlock(pos.south().east());
+    }
+
+    private boolean isSafeLandingSpot(BlockPos pos) {
+        BlockPos.MutableBlockPos mut = new BlockPos.MutableBlockPos(pos);
+        while (mut.getY() >= 0) {
+            IBlockState state = ctx.world().getBlockState(mut);
+            if (state.getMaterial().isLiquid()) { // lava
+                return false;
+            }
+            if (state.getMaterial().blocksMovement() && state.getBlock() != Blocks.MAGMA) {
```

Comment:
not quite right i think. magma should be in previous check. the way it's written it will skip magma entirely, while really it should treat magma like lava.

Reply:
actually nvm this entire thing is cringe

for example it doesn't consider fire properly

instead make it like:
if air { continue onward }
if netherrack { this is a valid landing spot }
if anything else { exit and return false }